src/hardware.c

```diff
@@ -413,8 +416,8 @@ void TempDetect() {
     static uint8_t init = 1;
     int16_t temp_new, temp_new0;
 
-    if (temp_tflg) {
-        temp_tflg = 0;
+    if (timer_2hz) {
```

Comment:
Check is useless here, just `timer_2hz = 0;`. Same below.

Reply:
`timer_2hz`will be 1 every about 1/2 second. 
`TempDetect()` and `LED_Task()` run based on this flag.

Reply:
Yes, currently it checks if the value is set (is equal to `timer_2hz != 0`) and set it to zero so that it is same if you just write it to zero without any additional check :)

Reply:
`timer_2hz` is used elsewhere.

Reply:
It is still the same behaviour whenever this method is called :)
Sounds like a possible bug to me... 🤔